CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+target_sources(mold-wrapper PRIVATE elf/mold-wrapper.c)
+
+add_executable(mold)
+target_compile_features(mold PRIVATE cxx_std_20)
+target_include_directories(mold SYSTEM PRIVATE third-party)
+
+if(NOT MSVC)
+    include(CheckLibraryExists)
+    check_library_exists(m pow "" LIBM_FOUND)
+
+    if(LIBM_FOUND)
+        target_link_libraries(mold PRIVATE m)
+    endif()
+
+    find_package(ZLIB REQUIRED)
+    target_link_libraries(mold PRIVATE ZLIB::ZLIB)
+endif()
+
+target_link_libraries(mold PRIVATE
+    TBB::tbb)
+
+if(MOLD_USE_MIMALLOC)
+    target_link_libraries(mold PRIVATE mimalloc::mimalloc)
+
+    if(MOLD_USE_SYSTEM_MIMALLOC)
+        target_compile_definitions(mimalloc INTERFACE USE_SYSTEM_MIMALLOC)
+    endif()
+endif()
+
+if(NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+    find_package(OpenSSL REQUIRED COMPONENTS Crypto)
+    target_link_libraries(mold PRIVATE OpenSSL::Crypto)
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES riscv64
+    OR CMAKE_SYSTEM_PROCESSOR MATCHES armv6)
+    find_library(ATOMIC_LIBRARIES atomic REQUIRED)
+    target_link_libraries(mold PRIVATE ${ATOMIC_LIBRARIES})
+endif()
+
+file(STRINGS .git/HEAD GIT_HASH)
+
+if(GIT_HASH MATCHES "ref: ")
+    string(REPLACE "ref: " "" GIT_HASH "${GIT_HASH}")
+    file(STRINGS .git/${GIT_HASH} GIT_HASH)
+endif()
+
+set_property(SOURCE main.cc macho/output-chunks.cc APPEND PROPERTY
+    COMPILE_DEFINITIONS MOLD_VERSION="${CMAKE_PROJECT_VERSION}")
+set_property(SOURCE main.cc APPEND PROPERTY
+    COMPILE_DEFINITIONS GIT_HASH="${GIT_HASH}")
+
+target_sources(mold PRIVATE
+    compress.cc
+    demangle.cc
+    filepath.cc
+    glob.cc
+    hyperloglog.cc
+    main.cc
+    multi-glob.cc
+    perf.cc
+    strerror.cc
+    tar.cc
+    uuid.cc
+    elf/arch-arm32.cc
+    elf/arch-arm64.cc
+    elf/arch-i386.cc
+    elf/arch-riscv64.cc
+    elf/arch-x86-64.cc
+    elf/cmdline.cc
+    elf/dwarf.cc
+    elf/gc-sections.cc
+    elf/icf.cc
+    elf/input-files.cc
+    elf/input-sections.cc
+    elf/linker-script.cc
+    elf/lto.cc
+    elf/main.cc
+    elf/mapfile.cc
+    elf/output-chunks.cc
+    elf/passes.cc
+    elf/relocatable.cc
+    elf/subprocess.cc
+    macho/arch-arm64.cc
+    macho/arch-x86-64.cc
+    macho/cmdline.cc
+    macho/dead-strip.cc
+    macho/input-files.cc
+    macho/input-sections.cc
+    macho/lto.cc
+    macho/main.cc
+    macho/mapfile.cc
+    macho/output-chunks.cc
+    macho/tapi.cc
+    macho/yaml.cc
+    third-party/rust-demangle/rust-demangle.c)
+
+# Create the ld and ld64 symlinks required for testing
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld symlink"
+    VERBATIM)
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld64
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld64 symlink"
+    VERBATIM)
+
+include(CTest)
```

Comment:
Please use the [module provided variable](https://cmake.org/cmake/help/latest/module/CTest.html#:~:text=module%20automatically%20creates%20a-,BUILD_TESTING,-option%20that%20selects) to guard everything testing related. This includes the symlink code above. [Example](https://github.com/friendlyanon/cmake-init-executable/blob/master/cmake/dev-mode.cmake#L3-L6).

Reply:
Done, thanks!